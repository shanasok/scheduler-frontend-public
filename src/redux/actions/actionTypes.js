export const LOAD_COURSES = "LOAD_COURSES";
export const SET_CURRENT_LESSON = "SET_CURRENT_LESSON";
export const SET_CURRENT_PAGE = "SET_CURRENT_PAGE";
export const SET_CURRENT_LESSON_CONTENT = "SET_CURRENT_LESSON_CONTENT";
export const SET_EDIT_SLIDE_TRUE = "SET_EDIT_SLIDE_TRUE";
export const SET_EDIT_SLIDE_FALSE = "SET_EDIT_SLIDE_FALSE";
export const SET_PAGE_CONTENT_EDITABLE = "SET_PAGE_CONTENT_EDITABLE";