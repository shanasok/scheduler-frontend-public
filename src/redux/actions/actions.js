import { LOAD_COURSES } from "./actionTypes";

// Actions are payloads of information that send data from your application to your store.

export const loadCourses = courses => ({
    type: LOAD_COURSES,
    payload: {
        courses: courses
    }
});
