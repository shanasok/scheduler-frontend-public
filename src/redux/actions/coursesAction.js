import { LOAD_COURSES } from "./actionTypes";

// Actions are payloads of information that send data from your application to your store.

export function setCoursesJson(coursesJson) {
    return {
        type: LOAD_COURSES,
        payload: {
            courses: coursesJson
        }
    }
}
