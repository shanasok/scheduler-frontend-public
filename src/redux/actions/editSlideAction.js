import { SET_EDIT_SLIDE_TRUE, SET_EDIT_SLIDE_FALSE } from "./actionTypes"

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setEditSlideTrue() {
    return {
        type: SET_EDIT_SLIDE_TRUE,
        payload: {
            setEditSlide: true
        }
    }
}

export function setEditSlideFalse() {
    return {
        type: SET_EDIT_SLIDE_FALSE,
        payload: {
            setEditSlide: false
        }
    }
}