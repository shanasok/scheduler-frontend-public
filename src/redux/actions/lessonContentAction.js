import { SET_CURRENT_LESSON_CONTENT } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentLessonContent(currentLessonContent) {
    return {
        type: SET_CURRENT_LESSON_CONTENT,
        payload: {
            currentLessonContent: currentLessonContent
        }
    }
}