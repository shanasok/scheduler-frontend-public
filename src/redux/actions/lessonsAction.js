import { SET_CURRENT_LESSON } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentLesson(currentLesson) {
    return {
        type: SET_CURRENT_LESSON,
        payload: {
            currentLesson: currentLesson
        }
    }
}
