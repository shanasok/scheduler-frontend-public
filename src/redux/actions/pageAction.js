import { SET_CURRENT_PAGE } from "./actionTypes";

// ActionlessonsAction.jss are payloads of information that send data from your application to your store.

export function setCurrentPage(currentPage) {
    return {
        type: SET_CURRENT_PAGE,
        payload: {
            currentPage: currentPage
        }
    }
}
