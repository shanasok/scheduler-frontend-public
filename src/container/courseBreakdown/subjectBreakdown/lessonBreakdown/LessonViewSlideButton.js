import React from 'react';
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { setCurrentLesson } from "../../../../redux/actions/lessonsAction";
import { setCurrentPage } from "../../../../redux/actions/pageAction"

//properties is the props that is passed in
export function setReduxLessonAndPage(properties){
    properties.actions.setCurrentLesson(properties.lesson);
    if (properties.lesson.pages[0])
        properties.actions.setCurrentPage(properties.lesson.pages[0])
}

class LessonViewSlideButton extends React.Component {

    render() {
        // return <Link to={{ pathname: "/lesson",  state:{ lesson: this.props.lesson } }}>
        const properties = this.props;
        return <Link to={{ pathname: "/lesson"}}>
            <button className="lessonViewButton ribbonSlideButton curvedButton"
                    onClick={() => setReduxLessonAndPage(properties)}>
                Slides
            </button>
        </Link>
    }
}

const mapStateToProps = state => ({

});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            setCurrentLesson,
            setCurrentPage,
        }, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LessonViewSlideButton)