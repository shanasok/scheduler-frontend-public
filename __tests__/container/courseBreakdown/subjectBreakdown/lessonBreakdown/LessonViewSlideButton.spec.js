import React from 'react';
import LessonViewSlideButton, { setReduxLessonAndPage } from '../../../../../src/container/courseBreakdown/subjectBreakdown/LessonBreakdown/LessonViewSlideButton';
import renderer from 'react-test-renderer';
import {Router} from "react-router-dom";
import { createMemoryHistory } from "history";
import {Provider as ReduxProvider} from "react-redux";
import configureMockStore from "redux-mock-store";
import {setCurrentLesson} from "../../../../../src/redux/actions/lessonsAction";
import {setCurrentPage} from "../../../../../src/redux/actions/pageAction";
import {mount, configure} from "enzyme";
import {MemoryRouter} from "react-router";
import Adapter from "enzyme-adapter-react-16";


var expect = require('expect');


const lesson = {
    id: 1,
    name: "test lesson name",
    lessonNumber: 1,
    pages: []
}

const mockStore = configureMockStore()
const mockState = {
    lesson: lesson
}
let store = mockStore(mockState)

it('renders correctly', () => {

    const history = createMemoryHistory();
    const lesson = {
        name: "Threading the Machine",
        lessonNumber: 1,
        pages:[],
    };
    history.location.state = {
        lesson: lesson
    }

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}><LessonViewSlideButton lesson={lesson}/></Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('populateLessonViewSlideButton', () => {
    it('should call both functions once', () => {
        const mockSetCurrentLesson = jest.fn(setCurrentLesson);
        const mockSetCurrentPage = jest.fn(setCurrentPage);
        const currentPage = {
            id: 1,
            pageNumber: 1,
        };
        const properties = {
            actions: {
                setCurrentLesson: mockSetCurrentLesson,
                setCurrentPage: mockSetCurrentPage,
            },
            lesson: {
                name: "Threading the Machine",
                lessonNumber: 1,
                pages: [currentPage],
            }
        };

        setReduxLessonAndPage(properties);
        expect(mockSetCurrentLesson.mock.calls.length).toBe(1);
        expect(mockSetCurrentPage.mock.calls.length).toBe(1);
    });

    it('should not set current page if there are no pages in the lesson', () => {
        const mockSetCurrentLesson = jest.fn(setCurrentLesson);
        const mockSetCurrentPage = jest.fn(setCurrentPage);
        const properties = {
            actions: {
                setCurrentLesson: mockSetCurrentLesson,
                setCurrentPage: mockSetCurrentPage,
            },
            lesson: {
                name: "Threading the Machine",
                lessonNumber: 1,
                pages: [],
            }
        };

        setReduxLessonAndPage(properties);
        expect(mockSetCurrentLesson.mock.calls.length).toBe(1);
        expect(mockSetCurrentPage.mock.calls.length).toBe(0);
    });

    const mockStore = configureMockStore()
    const mockState = {
        lesson: lesson
    }
    let store = mockStore(mockState)

    configure({adapter: new Adapter()});
    it('onClick of button it should call function one time', () => {
        let wrapper = mount(
            <MemoryRouter initialEntries={[ '/' ]}>
                <ReduxProvider store={store}>
                    <LessonViewSlideButton lesson={lesson}/>
                </ReduxProvider>
            </MemoryRouter>
        );
        wrapper.find('.lessonViewButton').simulate('click');

        expect(wrapper.find(LessonViewSlideButton)).toHaveLength(1);
        wrapper.unmount();
    });
});
