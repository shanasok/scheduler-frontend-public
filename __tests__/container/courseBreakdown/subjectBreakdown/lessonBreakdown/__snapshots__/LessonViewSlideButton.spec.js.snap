// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  href="/lesson"
  onClick={[Function]}
>
  <button
    className="lessonViewButton ribbonSlideButton curvedButton"
    onClick={[Function]}
  >
    Slides
  </button>
</a>
`;
